### PR TITLE
add support prompt release

### PR DESCRIPTION
**Pull request by abubakarsohail**

### CODE REVIEW

File: promptlayer/prompts/prompts.py [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
-)
+from promptlayer.utils import (get_api_key, promptlayer_get_prompt,
+                               promptlayer_publish_prompt)
 
 
-def get_prompt(prompt_name, langchain=False, version=None):
+def get_prompt(prompt_name, langchain=False, version=None, release=None):
     """
     Get a prompt template from PromptLayer.
     version: The version of the prompt to get. If not specified, the latest version will be returned.
```
Comment (Jped) [inline promptlayer/prompts/prompts.py:14]: @abubakarsohail can you add descriptions for `langchain` and `release` here? I guess we forgot to add langchain

File: promptlayer/utils.py [hunk 2]
```diff
@@ -179,7 +179,7 @@ def promptlayer_api_request_async(
     )
 
 
-def promptlayer_get_prompt(prompt_name, api_key, version=None):
+def promptlayer_get_prompt(prompt_name, api_key, version=None, release=None):
```
Comment (Jped) [inline promptlayer/utils.py:182]: @abubakarsohail is there any validation any where for adding both a `version` and a `release`? I guess that happens on the API end?

**Review by Jped — APPROVED** [review]
looks good!

File: promptlayer/utils.py (hunk 2 above)
Comment (abubakarsohail) [reply promptlayer/utils.py:182]: @Jped nope. I should probably add it over here to only accept a string for release.

Comment (Jped) [reply promptlayer/utils.py:182]: @abubakarsohail yes then please add validations.

File: promptlayer/prompts/prompts.py [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+def get_prompt(prompt_name, langchain=False, version: int = None, release: str = None):
     """
     Get a prompt template from PromptLayer.
+    prompt_name: the prompt name
+    langchain: Enable this for langchain compatible prompt
     version: The version of the prompt to get. If not specified, the latest version will be returned.
+    release: The specific release of a prompt you want to get. Setting this will supercede version
     """
+    try:
+        GetByName(prompt_name=prompt_name, version=version, release=release)
```
Comment (Jped) [inline promptlayer/prompts/prompts.py:24]: @abubakarsohail why is this class called `GetByName` if it isnt doing the actual get?

File: promptlayer/schemas/prompt.py [hunk 4]
```diff
@@ -100,3 +101,9 @@ class Base(BaseModel):
     ]
     prompt_template: PromptTemplate
     tags: List[str] = []
+
+
+class GetByName(BaseModel):
+    prompt_name: str
+    version: Union[PositiveInt, None] = None
```
Comment (Jped) [inline promptlayer/schemas/prompt.py:108]: @abubakarsohail we should copy the exact validation on API end, which includes accepting empty string

File: pyproject.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "promptlayer"
-version = "0.1.98"
+version = "0.1.99"
```
Comment (Jped) [inline pyproject.toml:3]: @abubakarsohail this needs to be upped again bc i pushed something

**Review by Jped — APPROVED** [review]
@abubakarsohail looks great. Just be sure we have this all tested as we are removing pydantic and updating get-prompt

**Pull request merged**